### DISCUSSION TITLE
Junior Dev here and lost a little

### DISCUSSION THREAD

**Discussion by umut1tr** [LostBeard/SpawnDev.BlazorJS#26] [General]
Hello everyone!

I'm currently facing a bit of a challenge with a new project I'm eager to start. As I embark on this journey, I realize there are aspects where I could benefit from the insights of experienced developers. My aim is to set up the project in a way that maximizes learning opportunities while ensuring a clean and efficient start.

The project's requirements are not overly complex, but I'm unsure if my proposed solution is the best approach. Here's what I aim to achieve:

Webcam Monitoring Application: I envision a program running on a single machine that captures feeds from three webcams. The application will periodically capture images from these webcams and utilize a "pixel diff" method to detect changes between successive images.

Integration with External API: Upon detecting changes, the application should promptly send the latest image containing the detected change to an external API. This API, hosted on a cloud server, will be responsible for performing calculations based on the received images and providing a response.

I aim to proceed as follows:

Develop a Blazor Web App utilizing .NET 8.

Employ the SpawnDev wrapper to initialize webcams on the client side. This will involve implementing a loop to monitor for changes in the webcam feeds. Upon detection of changes, an event will trigger a server-side client connection to my external API (which is part of future work).

I'm considering the possibility of implementing both client-side and server-side components. This decision arises from the need to integrate a dynamic UI, featuring a simple scoreboard. The scoreboard's values will be contingent on calculations, necessitating real-time updates in the UI without requiring a full page reload."

Is this suitable for what I want to do actually? Should I just make this a "Blazor Server App" ? I dont want to overcomplicate it.

Thanks for anyone who can give me a little feedback here.

Best regards,
Umi

**Comment by LostBeard** [comment]
Sounds like an interesting project. 

> Is this suitable for what I want to do actually? Should I just make this a "Blazor Server App" ? 

Personally, I would create a Blazor WASM client with an ASP.Net API server instead of a Blazor Server due to the processing you want to perform on the client However, either choice could work, but the Blazor Server route would likely require writing more Javascript, where a Blazor WASM project using SpawnDev.BlazorJS can be written entirely in C#.

You can check out [https://www.spawndev.com](https://www.spawndev.com) to see one of my Blazor WASM sites using SpawnDev.BlazorJS. It supports WebAuthn for login, video, audio, and text messaging between users, WebRTC file and folder sharing between a user's devices, 3D video and image viewing, and more.

Feel free to post any problems you come across here and I'll see if I can help.

Note. OpenCV may be able to help with the motion detection.

Looking forward to hearing about your progress on this. Best of luck!

**Reply by umut1tr** [reply]
Hi LostBeard,

Thank you very much for your response.

Do you have a Discord channel or any other platform where people experienced with your framework gather? It would be helpful to connect with others who have expertise in this area.

I'll do my best to tackle the task myself and will definitely reach out if I need further assistance. I really appreciate your offer—it's very kind of you!

I'm also considering incorporating a WASM Project and then adding an API Project on top of it. I plan to use SignalR or similar technology to update the UI when the calls return results.

To be honest, I don't think I'll need OpenCV for a custom detection. I have a TypeScript file that I want to work with, and I just need a simple "pixelmatch" to trigger some events. I've come across a framework called "mapbox/pixelmatch" for that purpose, and I'll give it a try once I've successfully set up BlazorJS in my WASM App and can access my camera feeds.

Best Regards,
Umi

**Comment by LostBeard** [comment]
> Do you have a Discord channel or any other platform where people experienced with your framework gather?

Not at this time. So far, using GitHub discussions and issues, along with StackOverflow has been sufficient.

**Comment by umut1tr** [comment]
I see thanks for your help. I finally managed to progess
________________________________
Von: Todd Tanner ***@***.***>
Gesendet: Freitag, 12. April 2024 00:30
An: LostBeard/SpawnDev.BlazorJS ***@***.***>
Cc: Umut ***@***.***>; Author ***@***.***>
Betreff: Re: [LostBeard/SpawnDev.BlazorJS] Junior Dev here and lost a little (Discussion #26)


Do you have a Discord channel or any other platform where people experienced with your framework gather?

Not at this time. So far, using GitHub discussions and issues, along with StackOverflow has been sufficient.
